Automattic-Tracks-iOS/Crash Logging/CrashLogging.swift

```diff
@@ -2,178 +2,251 @@ import Foundation
+        event.tags?["locale"] = NSLocale.current.languageCode
 
-        DDLogDebug("📜 Firing `shouldSendEvent`")
+        /// Always provide a value in order to determine how often we're unable to retrieve the value
+        event.tags?["app.state"] = ApplicationFacade().applicationState ?? "unknown"
 
         #if DEBUG
         DDLogDebug("📜 This is a debug build")
         let shouldSendEvent = UserDefaults.standard.bool(forKey: "force-crash-logging")
         #else
-        let shouldSendEvent = !CrashLogging.userHasOptedOut
+        let shouldSendEvent = !dataProvider.userHasOptedOut
         #endif
 
-        shouldSendEventCallback?(event, shouldSendEvent)
-
-        guard let eventLogging = self.eventLogging else {
-            DDLogDebug("📜 Cancelling log file attachment – Event Logging is not initialized")
-            return shouldSendEvent
+        /// If we shouldn't send the event we have nothing else to do here
+        guard shouldSendEvent else {
+            return nil
         }
 
-        guard let event = event else {
-            DDLogDebug("📜 Cancelling log file attachment – event is nil")
-            return shouldSendEvent
-        }
+        /// Read the current user from the Data Provider (though the Data Provider can decide not to provide it for functional or privacy reasons)
+        event.user = dataProvider.currentUser?.sentryUser
 
-        guard shouldSendEvent else {
-            DDLogDebug("📜 Cancelling log file attachment – should not send event")
-            return shouldSendEvent
+        /// Everything below this line is related to event logging, so if it's not set up we can exit
+        guard let eventLogging = self.eventLogging else {
+            DDLogDebug("📜 Cancelling log file attachment – Event Logging is not initialized")
+            return event
         }
 
         eventLogging.attachLogToEventIfNeeded(event: event)
 
-        return shouldSendEvent
-    }
-
-    /// The current state of the user's choice to opt out of data collection. Provided by the data source.
-    private static var userHasOptedOut: Bool {
-        /// If we can't say for sure, assume the user has opted out
-        guard let dataProvider = sharedInstance.dataProvider else { return true }
-        return dataProvider.userHasOptedOut
+        return event
     }
 
     /// Immediately crashes the application and generates a crash report.
-    public static func crash() {
-        Client.shared?.crash()
+    public func crash() {
+        SentrySDK.crash()
     }
 
-    public static var eventLogging: EventLogging? {
-        return sharedInstance.eventLogging
+    enum Errors: LocalizedError {
+        case unableToConstructAuthStringError
     }
 }
 
 // Manual Error Logging
 public extension CrashLogging {
 
-    /**
-     Writes the error to the Crash Logging system, and includes a stack trace.
+    ///
+    /// Writes the error to the Crash Logging system, and includes a stack trace.
+    /// - Parameters:
```

Comment:
```suggestion
    /// Writes the error to the Crash Logging system, and includes a stack trace.
    ///
    /// - Parameters:
```

Other headers in this file have a neat new line to separate description for parameters. This suggestion makes this consistent with them.